# CoolCorp

CoolCorp est vraiment très cool puisqu'ils ont bien voulu nous embaucher pour analyser leurs données suite au désastre du Titanic. Pour ce faire ils nous ont fourni un fichier train.csv, c'est ce fichier que nous allons devoir importer. 

Nous commencons par importer la librairie Panda puis par lire le fichier csv où header=(0) représente la ligne d'entête. 

In [57]:
import pandas as pd

# For .read_csv, always use header=0 when you know row 0 is the header row
df = pd.read_csv('train.csv', header=0)

cela nous permet de lire ce que nous avons dans notre fichier train.csv
Listons les premières lignes :


In [58]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Listons maintenant les types de ces données :

In [94]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Nous avons donc :
    - un PassengerId représentant l'identifiant du passager de type int64
    - Une indication de sa survivance au naufrage mais de type int64 et non bool
    - Une indication sur la classe de place occupée (int64) 
    - Un nom (objet)
    - Un sexe (objet)
    - Un age (floate64)
    - Une indication sur un lien de fratrie avec d'autres passagers (int64)
    - Une indication sur un lien de parenté avec d'autres passagers (int64)
    - Une indication sur le N° de ticket (objet)
    - un prix du billet (float64)
    - Une indication sur le N° de cabine (objet)
    - une indication sur le lieu d'embarquement (objet) avec C = Cherbourg; Q = Queenstown; S = Southampton

1) Commençons par compter notre nombre de passager. 
    Nous mettons toutes nos données dans un objet data_all (en recopiant bêtement le code du prof, je sais c'est mal :p  )
    La dernière ligne du paragraphe suivant affiche le résultat, elle comporte toutefois un symbole étrange : 
    {p:.2f} --> kesako ?


In [95]:
data_all = []
with open('train.csv') as train_file:
    csv_reader = csv.reader(train_file, delimiter=',', quotechar='"')
    for row in csv_reader:
        data_all.append(row)
data_all = np.array(data_all)
data = data_all[1::]
number_passengers = np.size(data[0::,1].astype(np.float))
print ("le nombre de passager est de {p:.2f}" .format (p=number_passengers))

le nombre de passager est de 891.00


2) Comptons maintenant le nombre de femmes et d'hommes

In [14]:
nbFemmes = data[0::,4] == "female"
nbHommes = data[0::,4] != "female"

In [66]:
totalFemmes = np.sum(nbFemmes).astype(np.float)
totalHommes = np.sum(nbHommes).astype(np.float)

In [67]:
print('total femmes {p:.2f}'.format(p=totalFemmes))

total femmes 314.00


In [68]:
print('total hommes {p:.2f}'.format(p=totalHommes))

total hommes 577.00


Nous pouvons constater ici fièrement que 577 + 314 = 891, une première victoire !

3) Calculons maintenant la proportion d'hommes et de femmes à bord du bateau
    La proportion d'homme = nombre d'hommes / nombre total de passagers

In [69]:
pHommes = (totalHommes / number_passengers).astype(float)
print('Proportion d homme  {p:.2f}'.format(
        p=pHommes))

Proportion d homme  0.65


In [71]:
pFemmes = totalFemmes / number_passengers
print('Proportion de femme  {p:.2f}'.format(
        p=pFemmes))

Proportion de femme  0.35


Là encore nous constatons que 0.35 + 0.65 = 1, ce qui nous fait 2 victoires !

Nous constatons alors que la proportion des hommes à bord du bateau est bien supérieure à la proportion de femmes parmi le nombre total de passagers. 

# Etude des survivants

En route vers une 3ème victoire... ou pas :p

Nous allons donc étudier les survivants Pour cela nous allons :

1) Déterminer le nombre de survivants 


In [85]:
Survivants = np.sum(data[0::2] == '1')

In [86]:
print('le nombre de survivants est de {p:.2f}'.format( p=Survivants))

le nombre de survivants est de 426.00


In [98]:
type(Survivants)

numpy.int32

In [99]:
type(number_passengers)

int

2) calculons maintenant la proportion de survivants :
    = nombre de survivants par rapport au nombre de passagers total
    = Survivants / number_passengers
   

In [118]:
pS = (Survivants / number_passengers).astype(float)
print('la proportion de survivants est de {p:.2f}'.format(p=pS))

la proportion de survivants est de 0.00


In [110]:
nbSurvivants = Survivants.astype(float)
print('le nombre de survivants est de {p:.2f}'.format( p=nbSurvivants))

le nombre de survivants est de 426.00


In [111]:
number_passengers.astype(float)

AttributeError: 'int' object has no attribute 'astype'

In [119]:
pSurvivants = Survivants.astype(float) / number_passengers.astype(float)
print('le nombre de survivants est de {p:.2f}'.format( p=pSurvivants))

AttributeError: 'int' object has no attribute 'astype'

In [113]:
print('la proportion de survivants est {p:.2f}'.format( p=pSurvivants))

la proportion de survivants est 0.00


Ici nous avons clairement une erreur puisque nous savons qu'elle est d'environ 1/2

Les différents tests montrent qu'il s'agit d'une erreur de type de variable. 

Afin de palier à ce problème continuons en ommettant l'obstacle du langage Python.

# Démarche mathématique

Après avoir calculé la proportion de survivants il faudrait calculer la proportion de femmes survivantes et vérifier qu'elle est égale à la proportion totale de femmes présentes sur le bateau. On saurait ainsi si la probabilité de survivre est liée au sexe.

on aurait donc les calculs suivants :
- nbFS = nombre de femmes survivantes
- pFS = probabilité de survivre en sachant qu'on est une femme (noté : P(F|S)) = (p(F) inter p(S)) / p(S)

puis on comparerait cette probabilité avec la probabilité de survie totale (= p(S), celle que nous n'avons pas réussi à calculer).
Si le nombre est identique alors les 2 variables sont indépendantes et nous n'avons pas de lien entre la probabilité de survivre avec la probabilité d'être une femme. 
Parions que la probabilité de survivre en étant une femme est supérieure à la probabilité totale...

Cela nous permettrait de poser le prédicat que la probabilité de survie est supérieure lorsqu'on est une femme que lorsqu'on est un homme, hypothèse sommes toutes, très banale ! (bin koi ? pour un coup que les femmes ont un avantage sur les hommes...)

Certes il reste encore bien des notions à apprendre, calculer la moyenne qui n'a pas de sens dans cette hypothèse, la variance par rapport à la moyenne, l'écart-type, le coefficient de corrélation et la regression linéaire ou la régression logistique suivant l'hypothèse émise ou le type de données. 
Nous n'avons pas tout perdu pour autant : nous avons appris utiliser github, des notions mathématiques et des notions de python en quelques jours seulement.

Merci @Jeff Abrahamson de nous torturer ainsi. 
